File: src/chain-observers/classifiers/impacted-account-classifier.ts
import { CollectorClassifierBase } from "./collector-classifier-base";
import type { IOperationBaseData } from "./operation-classifier";

export interface IImpactedAccount extends IOperationBaseData {
  name: string;
}

export interface IImpactedAccountData {
  impactedAccounts: Record<string, IImpactedAccount>;
}

export class ImpactedAccountClassifier extends CollectorClassifierBase<IImpactedAccountData> {}

File: python/workerbee/chain_observers/classifiers/impacted_account_classifier.py
from __future__ import annotations

from ..classifier_results import ImpactedAccountClassifierData
from .collector_classifier_base import CollectorClassifierBase


class ImpactedAccountClassifier(CollectorClassifierBase[ImpactedAccountClassifierData]):
    """IImpactedAccountData:
      impacted_accounts: dict[str, IImpactedAccount]

    IImpactedAccount:
      name: str
      operations: list[IOperationTransactionPair]
    """

    pass
